routing/missioncontrol.go

```diff
@@ -299,6 +305,46 @@ func generateBandwidthHints(sourceNode *channeldb.LightningNode,
 	return bandwidthHints, nil
 }
 
+// edgePolicyDirection returns the direction of the edge policy in the format used
+// by edge locators (i.e. 1 if lnwire.ChanUpdateDirection is set in flags)
+func edgePolicyDirection(ep *channeldb.ChannelEdgePolicy) uint8 {
+	if ep.Flags&lnwire.ChanUpdateDirection == 0 {
+		return 0
+	}
+	return 1
+}
+
+// updateEdgePolicy updates the channel edge policy parameters
+func updateEdgePolicy(up *lnwire.ChannelUpdate,
+	ep *channeldb.ChannelEdgePolicy) {
+
+	ep.FeeBaseMSat = lnwire.MilliSatoshi(up.BaseFee)
+	ep.FeeProportionalMillionths = lnwire.MilliSatoshi(up.FeeRate)
+	ep.TimeLockDelta = up.TimeLockDelta
+	ep.MinHTLC = lnwire.MilliSatoshi(up.HtlcMinimumMsat)
+}
+
+// UpdateEdgePolicy updates edge policy of the routing hints kept in the
+// payment session.
+func (p *paymentSession) UpdateEdgePolicy(el *edgeLocator,
+	update *lnwire.ChannelUpdate) {
+
+	log.Debugf("Updating edge %v policy in Mission Control", el)
+
+	// Run over hints and update policy if edge id and direction match.
+	for _, edges := range p.additionalEdges {
+		for _, edge := range edges {
+			if edge.ChannelID != el.channelID {
+				continue
+			}
+			if edgePolicyDirection(edge) != el.direction {
```

Comment:
We could also add a method for creating `edgeLocator` from a `ChannelUpdate`.

Reply:
Added